### ISSUE TITLE
Nmon plugin

### ISSUE THREAD

**Issue by dadolab** [influxdb/telegraf#444]
Hello

did you foreseen a nmon plugin?

Thanks

**Comment by sparrc** [comment]
Nope, but what do you need it for? You should be able to get all the info in nmon using Telegraf's other system plugins

**Comment by dadolab** [comment]
I have several AIX and linux server that are collecting system metric using, I would like to collect this metric in influxdb

**Comment by sparrc** [comment]
@dadolab would you like to collect it from an nmon csv file? Or run nmon at collection time and parse stdout?

**Comment by dadolab** [comment]
It would be nice to have it at collection time to have online, but also have it collected daily from nmon file/csv will be enough

But for AIX will be possible to have telegraf binary?

**Comment by sparrc** [comment]
@dadolab I don't think this will be possible for a loooooong time, Go doesn't support AIX: https://github.com/golang/go/issues/7208

**Comment by mark-rushakoff** [comment]
Until a plugin is written... there's [a standalone tool](https://github.com/adejoux/nmon2influxdb) out there that takes an nmon file and imports it into InfluxDB. I haven't used it myself but I've stumbled across it while curating [awesome-influxdb](https://github.com/mark-rushakoff/awesome-influxdb).

**Comment by dadolab** [comment]
@mark-rushakoff: I'm already using nmon2influxdb, but the "upload" is manual/offline

**Comment by thaentzler** [comment]
Seems like golang is available for aix now  : https://www.ibm.com/developerworks/community/wikis/home?lang=en#!/wiki/W51a7ffcf4dfd_4b40_9d82_446ebc23c550/page/Go%20compilers%20on%20Power

So maybe things will move. But difficulty here is that aix doesn't really work like linux.... So standard sys plugin will not work as expected.

**Comment by dadolab** [comment]
it looks like it will be available only for linux on power, not yet for AIX (and unfortunately no info about it...)

**Comment by freebsdly** [comment]
you can use a script to report nmon file to telegraf http listener plugin

**Comment by toni-moreno** [comment]
Here a new and rewritted version of the nmon2influxdb, for AIX and nmon based info with ssh access.

https://github.com/adejoux/pSeriesCollector

**Comment by russorat** [comment]
I believe there is also a tool called `nimon` available that reports nmon stats directly to influxdb via line protocol. See http://nmon.sourceforge.net/pmwiki.php?n=Site.Njmon for more info.

**Comment by toni-moreno** [comment]
This tool (https://github.com/adejoux/pSeriesCollector)  read PSeries HMC stats and also does real time remote "tail -f"  via ssh and parse nmon files. 

I think ..it could work without lpar autodiscovery and only does remote nmon parser from any vm ( lpar or not lpar)

**Comment by nigelargriffiths** [comment]
I am the developer of the nmon and njmon/nimon mentioned above - nimon is meant to be the replacement for nmon and has many time the number of stats. It also works in the Virtual I/O Server (AIX based) and captures virtual devices.
nimon output InfluxDB Line Protocol direct to InfluxDB or Telegraf.  njmon output JSON data and can be used with a njmond.py central collection point to save the JSON on inject in to InfluxDB.  ON a small virtual machines it collects ~1500 stats.  I think we have AIX covered.
If not let me know. My email can be found on the open source project website: http://nmon.sourceforge.net/pmwiki.php?n=Site.Njmon

**Comment by powersj** [comment]
Hi,

Thanks for the interest in this plugin. Per the previous comment, given the enormous amount of data points that nmon/nimon/njmon collect, and the availability of njmon to output to JSON data, I will close this request.

Users who use this tool can output to JSON and then use Telegraf to parse the JSON and send it to their output of choice. 

Thanks!
